test/tools/jsontests/BlockChainTests.cpp

```diff
@@ -117,304 +120,380 @@ void doBlockchainTestNoLog(json_spirit::mValue& _v, bool _fillin)
 		//Select test by name if --singletest is set and not filling state tests as blockchain
 		if (!Options::get().fillchain && !TestOutputHelper::passTest(testname))
 		{
-			o.clear(); //don't add irrelevant tests to the final file when filling
-			continue;
+			if (_fillin)
```

Comment:
1. `first with _fillin = true` does this suggest, `second time with _fillin == false`?
2. why cannot one select tests by testname, when this function is executed from statetests?  Can't we fill tests first and write the filled statetest to the disk, and then fill chain?